cms/utils/log_operations.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+
+from django.contrib.admin.models import LogEntry, ADDITION, CHANGE, DELETION
+from django.contrib.contenttypes.models import ContentType
+from django.utils.translation import ugettext_lazy as _
+
+
+"""
+Logs are all collected here to keep the messages and contents consistent across all operations.
+May seem repetitive but allows each log to be unique for different circumstances.
+Method names make the log being created clear and concise
+
+Issues with current solution:
+    - A log entry isn't created when a page is created using the new page wizard. Again showing a flaw with testing via the API and not the actual fn's!!
+    - Difficult to find the log page creation in the current solution!! The existing implementation,uses the LogEntry model directly, others use the ModelAdmin Implementation. There should be one way where possible!!
+
+Design Considerations:
+    - Allow plugins to integrate with the CMS logger??
+    - A log helper method "create_log" that uses LogEntry and allows the removal of LogEntry at a later date.
+    - A page move event is recorded as CHANGE with no message. A "Moved." message has been manually added
+    - Django admin keeps all logs in one area rather than spreading them out and potentially having them set inconsistently and scattered in multiple files.
+"""
+
+
+"""
+Log helper
+"""
+
+
+def create_log(user_id, content_type_id, object_id, object_repr, action_flag, change_message):
+    """
+    Helper method
+    Although this function is repetitive it allows external plugins to access the same logging mechanisms as the cms
```

Comment:
To me this seems more like a comment than a docstring :) A docstring would usually describe what the function does, for example "Logs changes to an object".

Reply:
I'm not sure if the method will stay. Depends on Paulos design choices on how this will be done. May be gone in my final PR.